kotest-assertions/kotest-assertions-shared/src/commonMain/kotlin/io/kotest/assertions/NondeterministicHelpers.kt

```diff
@@ -0,0 +1,19 @@
+package io.kotest.assertions
+
+typealias SuspendingPredicate<T> = suspend (T) -> Boolean
+
+typealias SuspendingProducer<T> = suspend () -> T
+
+interface NondeterministicListener<in T> {
+   suspend fun onEval(t: T)
+
+   companion object {
+      val noop = object : NondeterministicListener<Any?> {
+         override suspend fun onEval(t: Any?) { }
```

Comment:
Add an argument here that passes an object representing internal state like how many iters and current runtime